test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -279,6 +279,45 @@ struct test_transform_iterator {
     }
 };
 
+struct test_permutation_iterator
+{
+    template <typename T1, typename T2>
+    void
+    operator()(::std::vector<T1>& in1, ::std::vector<T2>& in2)
+    {
+        T1 iota_max = ::std::numeric_limits<T1>::max() < in1.size() ? ::std::numeric_limits<T1>::max() : in1.size();
+        ::std::iota(in1.begin(), in1.begin() + iota_max, T1(0));
+        ::std::reverse_copy(in1.begin(), in1.begin() + iota_max, in2.begin());
```

Comment:
`test_zip_iterator` calls sort to check that the comparison on tuple works correctly.  I think it's important to call a simple algorithm using `permutation_iterator` as well to ensure the indirection in the mapping is working as expected.

Reply:
I know about `test_zip_iterator`...  But. in these cases we have got some kind of implicit testing.. with no guarantee the full teste coverage for the iterators... And looking into this iterator test we even cannot estimate test coverage because we dont "known" / "see" what iterator operations are used within an implementation of the algo... and that implementation may be changed in the future.

Reply:
`test_random_iterator` is called after the algorithms to explicitly test the iterator operations. I've added explicit tests of the constructor used in `make_permutation_iterator` and for the `base` and `map` methods.  I think this covers all methods of the iterator explicitly.